### PR TITLE
Mixed: crash fix, doc, doc memory/speed trade-off, fix tests

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: doc/tutorial/using_gpu.txt [hunk 1]
```diff
@@ -783,5 +783,9 @@ Modify and execute to support *stride* (i.e. to avoid constraining the input to
 Note
 ----
 
-See :ref:`example_other_random` to know how to handle random numbers
-on the GPU.
+* See :ref:`example_other_random` to know how to handle random numbers
+  on the GPU.
+
+* The mode `FAST_COMPILE` disable c code, so also disable the GPU. You
```
Comment (abergeron) [inline doc/tutorial/using_gpu.txt]: disable c code -> disables C code
also disable the -> also disables the

File: doc/tutorial/using_gpu.txt [hunk 2]
```diff
@@ -783,5 +783,9 @@ Modify and execute to support *stride* (i.e. to avoid constraining the input to
...
+  can use the Theano flag optimizer='fast_compile' to speed up
+  compilatino and keeping the GPU.
```
Comment (abergeron) [inline doc/tutorial/using_gpu.txt]: compilatino and keeping -> compilation and keep

File: theano/compile/function_module.py [hunk 3]
```diff
@@ -560,29 +560,34 @@ def copy(self, share_memory=False, swap=None, delete_updates=False,
...
-            Function. Otherwise, it will be old + " copy"
-
-            profile -- as theano.function profile parameter
-        ---------------------
-        Returns:
-            func -- Copied theano.Function
+
+        Parameters
+        ----------
+            share_memory : boolean
```
Comment (abergeron) [inline theano/compile/function_module.py]: This shouldn't be indented with regards to the header

File: theano/gof/graph.py [hunk 4]
```diff
@@ -451,6 +451,31 @@ def eval(self, inputs_to_values=None):
...
+
+        Notes
+        -----
+
+        `eval` will be slow the first time you call it on a variable --
+        it needs to call :func:`function` to compile the expression behind
+        the scenes. Subsequent calls to :func:`eval` on that same variable
+        will be fast, because the variable caches the compiled function.
+
+        This way of computing have more overhead then a normal Theano
```
Comment (abergeron) [inline theano/gof/graph.py]: have more overhead then -> has more overhead than

File: theano/gof/graph.py [hunk 5]
```diff
@@ -451,6 +451,31 @@ def eval(self, inputs_to_values=None):
...
+        function, so don't use it too much in real script. But is
```
Comment (abergeron) [inline theano/gof/graph.py]: in real script. -> in real scripts.

Also I would just delete the last sentence.

File: theano/sandbox/tests/test_blocksparse.py [hunk 6]
```diff
@@ -212,8 +212,10 @@ def metaop(b, h, W):
         def op(b, h, W):
             return self.gemv_op(b.take(oIdx, axis=0), W, h, iIdx, oIdx)
 
-        utt.verify_grad(metaop, [b_val, h_val, W_val], mode=self.mode)
-        utt.verify_grad(op, [b_val, h_val, W_val], mode=self.mode)
+        eps = 3e-4
+        eps = 3e-3
```
Comment (abergeron) [inline theano/sandbox/tests/test_blocksparse.py]: Delete one or the other.

File: theano/tensor/nnet/sigm.py [hunk 7]
```diff
@@ -314,6 +315,9 @@ def values_eq_approx_remove_low_prec(a, b):
 
 
 class ScalarSoftplus(scalar.UnaryScalarOp):
+    """
+    This help numerical stability.
```
Comment (abergeron) [inline theano/tensor/nnet/sigm.py]: help -> helps

**Comment by abergeron** [comment]
Looks ok apart from some documentation typos/format.

**Comment by nouiz** [comment]
updated the PR with fix from the code review.

**Comment by abergeron** [comment]
Ok to merge when travis passes.

**Pull request merged**